handler/src/main/java/io/netty/handler/ssl/KeyManagerConfiguration.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.ssl;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+import java.io.File;
+import java.io.InputStream;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collections;
+
+/**
+ * Configures the KeyManager for an SslContextBuilder.
+ *
+ * If the instance is not for configuring a server, key and keyCertChain may be left unspecified,
+ * which disables mutual authentication.
+ *
+ * If the key is not password protected, the password may be left unspecified.
+ */
+public final class KeyManagerConfiguration {
+
+    private PrivateKey key;
+    private X509Certificate[] keyCertChain;
+    private String keyPassword;
+
+    /**
+     * Creates a new instance.
+     */
+    public KeyManagerConfiguration() { }
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1186)